task-service/presubmit-pr.ts

```diff
@@ -0,0 +1,34 @@
+import * as fetch from 'node-fetch';
+import * as functions from 'firebase-functions';
+
+import {setStatusOnGithub} from '../shared/github';
+
+/** 
+ * The handler function for a presubmit pr task.
+ * 
+ * Checks the if the pr is able to be cherry-picked cleanly to its target branch, and
+ * set the status on github as a result.
+ */
+export const handlePresubmitPrTask = functions
+  .pubsub
+  .topic('presubmit-pr-task')
+  .onPublish(message => {
+    const {org, repo, pr, branch, sha} = message.json;
+    const url = `https://presubmit-service-dot-branch-manager.appspot.com/check_pr`;
+    const config = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({org, repo, pr, branch, sha})
+    }
+    fetch(url, config).then(response => response.json()).then(
```

Comment:
How about using `async/await` here as well?

Reply:
If we use `async/await` here, we have to put it in a try catch to handle the failure case.  Its seems weird since its not failing it just is getting back a failing result from the presubmit service.

Thoughts?